### PR TITLE
Setup depends

### PR DESCRIPTION
**Pull request by phadej**
Let's see how it goes with travis

### CODE REVIEW

File: bytes.cabal [hunk 1]
```diff
@@ -30,6 +30,10 @@ source-repository head
   type: git
   location: git://github.com/ekmett/bytes.git
 
+custom-setup
+  setup-depends:
+    Cabal < 1.26
```
Comment (RyanGlScott) [inline bytes.cabal]: Why do you need `custom-setup` for this trick to work?

**Comment by ekmett** [comment]
This is an attempt towards getting us a proper custom-setup stanza

**Comment by bgamari** [comment]
For what it's worth development snapshots of the `Cabal` library will soon infer a `Cabal <1.25` `setup-depends` constraint where one is not explicitly provided. This is merely a temporary measure until the in-tree version number is bumped; when the final `Cabal` 2.0 is released the implicit constraint will be `Cabal <2.0`. See https://github.com/haskell/cabal/pull/3939 for details.

**Comment by phadej** [comment]
This is obsolete

**Pull request closed without merging**